### PR TITLE
Alpha version of containerized benchmark

### PR DESCRIPTION
**Pull request by willgaviria**
# Summary

We've modified extensively from the original fork from `harvard-edge/dataperf-speech-example` to create the source code for the containerized version of the training data selection for visual classification benchmark (alpha version).

Briefly:
- `main.py` is the primary function to generate the scores for this benchmark
- `utils.py` provides supporting functions for `main.py` to allow misc tasks such as loading and saving
- `eval.py` provides key eval functions for `main.py` to allow for model training and scoring
- `constants.py` stores important constants used by the scripts to make the code more extensible
- `task_setup.yaml` serves as a configuration file for the submission to be edited by the user
- `Dockerfile` and `docker-compose.yaml` enable containerization of the benchmark tasks for scoring standardizatio
- A template file system for storing downloaded resources, submission training sets and generated scores can be found in `data/`
- This template file system also includes `data/train_sets/random_500.csv` (a test training set) and `data/results/results_for_random_500.csv` (its corresponding results) for testing of installation

# Todo
Update readme to reflect these changes

### CODE REVIEW

File: eval.py [hunk 1]
```diff
@@ -1,49 +1,35 @@
...
-        [np.ones(len(train_target_vectors)), np.zeros(len(train_nontarget_vectors))]
-    )
-
-    clf = sklearn.linear_model.LogisticRegression(random_state=0).fit(train_x, train_y)
-    print("eval score", clf.score(eval_x, eval_y))
-
-
-if __name__ == "__main__":
-    fire.Fire(main)
+from typing import Optional
```
Comment (coactive-grasshopper) [inline eval.py:1]:
```suggestion
from typing import Dict, Optional
```

**Comment by coactive-grasshopper** [comment]
This looks good to me

File: eval.py (hunk 1 above)
Comment (willgaviria) [reply eval.py:1]: Done

File: eval.py [hunk 2]
```diff
@@ -1,49 +1,35 @@
...
+        random_state=c.RANDOM_SEED)) -> BaseEstimator:
+
+    df = df.select(c.LABEL_COL, c.EMB_COL).toPandas()
+    X = df[c.EMB_COL].values.tolist()
+    y = df[c.LABEL_COL].values.tolist()
+
+    return clf.fit(X, y)
+
+
+def score_classifier(df: DataFrame, clf: BaseEstimator) -> dict:
```
[the remark this answers is not in the document]
Comment (willgaviria) [reply eval.py:24]: Done

**Pull request merged**